### ISSUE TITLE
Parallelize SS.py and TPI.py

### ISSUE THREAD

**Issue by evan-magnusson** [open-source-economics/OG-USA#56]
There are multiple ways we can parallelize SS.py and TPI.py.

SS.py:
      In the SS_solver() function, the for loop {for j in xrange(J)} on line 157 can be parallelized, as each j index of the array does not rely on anything that happens before it.  J is 7 right now, and so the speedup wouldn't be huge right now.  However, this would allow us to easily set J=100, and not have massive slowdowns.  Also, if there is a way to parallelize the minimizer, huge speed ups would be had.  The minimizer tests each element of the gradient (there are 87 elements to test), jumps, tests each element, jumps, etc.  However, testing each of the elements does not require any info from testing the previous element.  So it could test each element in parallel, jump, test each element in parallel, jump, etc.  This would be much faster.  So we either need to build a minimizer or find one that is parallelized.

TPI.py:
     In the while loop that starts on line 275, there are a couple for loops that can be parallelized.  The easiest would be the for loop on line 287 {for j in xrange(J)}.  As in SS.py, this can be parallelized.  Within this for loop (for each j), there are two for loops (not nested, separate) on line 288 and 299.  These can happen at the same time, and these loops themselves can be parallelized for each s and t.  So, we can parallize the J for loop, and then each processor can spawn 2 processes, for the S-2 loop and the T loop.  Then, those can also spawn S-2 or T processes, respectively.  We might not want to parallelize this much, as this is hundreds of processors.  We might want to just parallelize the T for loop, as T is the largest.  But we have lots of options here.

**Comment by rickecon** [comment]
@talumbau said he would work on this.

**Comment by jdebacker** [comment]
Superseded by Issue #224
